pkg/oci/puller/puller.go

```diff
@@ -83,7 +91,37 @@ func (p *Puller) Pull(ctx context.Context, artifactType oci.ArtifactType, ref, d
 			repo.Reference.Repository, repo.Reference.Reference, repo.Reference.Repository, err)
 	}
 
+	descReader, err := repo.Fetch(ctx, desc)
```

Comment:
```suggestion
	descReader, err := localTarget.Fetch(ctx, desc)
```
We have already fetched the graph from the remote repo, we can get the descriptor from the local store to avoid fetching it from the remote repo.

Reply:
Yes, you are super right!